wallets/core/src/namespaces/common/hooks/changeAccountSubscriber.ts

```diff
@@ -0,0 +1,112 @@
+import type {
+  Actions,
+  SubscriberCleanUp,
+} from '../../../hub/namespaces/types.js';
+import type { Subscriber } from '../../../mod.js';
+import type {
+  AutoImplementedActionsByRecommended,
+  SupportedProviderTypes,
+} from '../types.js';
+
+type ChangeAccountSubscriberBuilderConfig<
+  EventType,
+  ProviderAPI extends SupportedProviderTypes,
+  AddEventListenerReturnType
+> = {
+  getInstance: () => ProviderAPI;
+  format: (instance: ProviderAPI, event: EventType) => Promise<string[]>;
+  shouldItDisconnect: (event: EventType) => boolean;
+  addEventListener: (
+    instance: ProviderAPI,
+    callback: (event: EventType) => void
+  ) => AddEventListenerReturnType;
+  removeEventListener: (
+    instance: ProviderAPI,
+    callback: (event: EventType) => void,
+    addEventListenerReturn: AddEventListenerReturnType
+  ) => void;
+};
+export class ChangeAccountSubscriberBuilder<
+  EventType,
+  AddEventListenerReturnType,
+  ProviderAPI extends SupportedProviderTypes,
+  ActionsType extends Actions<ActionsType> &
+    Actions<AutoImplementedActionsByRecommended>
+> {
+  private config = {} as ChangeAccountSubscriberBuilderConfig<
+    EventType,
+    ProviderAPI,
+    AddEventListenerReturnType
+  >;
+  setGetInstance(getInstance: (typeof this.config)['getInstance']) {
+    this.config.getInstance = getInstance;
+    return this;
+  }
+  setFormat(format: (typeof this.config)['format']) {
+    this.config.format = format;
+    return this;
+  }
+  setShouldItDisconnect(
+    shouldItDisconnect: (typeof this.config)['shouldItDisconnect']
+  ) {
+    this.config.shouldItDisconnect = shouldItDisconnect;
+    return this;
+  }
+  setAddEventListener(
+    addEventListener: (typeof this.config)['addEventListener']
+  ) {
+    this.config.addEventListener = addEventListener;
+    return this;
+  }
+  setRemoveEventListener(
+    removeEventListener: (typeof this.config)['removeEventListener']
+  ) {
+    this.config.removeEventListener = removeEventListener;
+    return this;
+  }
+  build(): [Subscriber<ActionsType>, SubscriberCleanUp<ActionsType>] {
+    for (const configKey in this.config) {
+      if (!this.config[configKey as keyof typeof this.config]) {
+        throw new Error(`"${configKey}" has not been set`);
```

Comment:
```suggestion
        throw new Error(`Required "${configKey}" operation has not been set for "changeAccountSubscriber"`);
```
Errors can be more informative to help make debugging easier.